share/wake/lib/core/list.wake

```diff
@@ -802,3 +835,30 @@ export def groupBy (cmp: a => a => Boolean) (list: List a): List (List a) =
             (elem, head, rest), tail
         _ = (elem, Nil), acc
     foldr combo Nil sorted
+
+export def optionToList: Option a => List a = match _
+    Some x = x, Nil
+    None = Nil
+
+export def editFilter (selectFn: a => Boolean) (editFn: a => a): List a => List a =
+    def editIf x = if selectFn x then editFn x else x
+    map editIf
+
+export def lookup (selectFn: a => Boolean): List (Pair a b) => List b =
+  def select (Pair key value) = if selectFn key then Some value else None
+  mapPartial select
+
+export def editLookup (selectFn: a => Boolean) (editFn: b => b): List (Pair a b) => List (Pair a b) =
+  editFilter (selectFn _.getPairFirst) $ editPairSecond $ editFn
+
+# TODO: Do we want Option versions of these?
```

Comment:
It would be best to address this before merging.  For my part, I'm not sure we need these in an `Option` sequence; `expectFirst` is just `head`, `expectMany` would just be `Some`, and `expectOne` is easy enough to handle through a `match` or `require` if it ever comes up.

Reply:
I mean, I still think it's more idiomatic to use an `else` with the `require` rather than terminating the pipeline with `| expectXYZ`, but I'm willing to see how it plays out in practice.

Reply:
Ok lets remove that comment then and not add the option versions of `expectOne` just yet